relayer/src/merkle_roots/mod.rs

```diff
@@ -141,16 +143,27 @@ pub struct MerkleRootRelayer {
     waiting_for_authority_set_sync: BTreeMap<u64, Vec<GearBlock>>,
 
     save_interval: Interval,
+
+    main_interval: Interval,
+    spike: SpikeConfig,
+    first_pending_timestamp: Option<Instant>,
+
+    queued_root_timestamps: VecDeque<Instant>,
+    merkle_root_batch: Vec<PendingMerkleRoot>,
 }
 
 impl MerkleRootRelayer {
     pub async fn new(
         api_provider: ApiProviderConnection,
         storage: Arc<MerkleRootStorage>,
+        spike_config: SpikeConfig,
     ) -> MerkleRootRelayer {
         let mut save_interval = tokio::time::interval(Duration::from_secs(60));
         save_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
 
+        let mut main_interval = tokio::time::interval(Duration::from_secs(15));
```

Comment:
the same question: constant or a CLI-argument? And for above `save_interval` too